File: acceptance/rls129/71_simplify_path.rs
impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut files = path.split('/');
        let mut filenames = Vec::new();
        while let Some(f) = files.next() {
            if f == "." {}
            else if f == ".." {
                filenames.pop();
            } else if f == "" {}
            else {
                filenames.push(f);
            }
        }
        "/".to_owned() + &filenames.join("/")
    }
}

File: acceptance/csangam/71 - simplify path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        final=[]
        splitted=path.split('/')
        for i in splitted:
            if i!='' and i!='.' and i!='..':
                final.append(i)
            elif i=='..':
                if len(final)>0:
                    final.pop()

        return '/'+'/'.join(final)

